programs/ziti-edge-tunnel/netif_driver/linux/tun.c

```diff
@@ -130,12 +144,56 @@ static int run_command(const char *cmd, ...) {
     return r;
 }
 
-static int run_command_ex(bool log_nonzero_ec, const char *cmd, ...) {
-    va_list args;
-    va_start(args, cmd);
-    int r = run_command_va(log_nonzero_ec, cmd, args);
-    va_end(args);
-    return r;
+static int is_systemd_resolved_acquired_name(sd_bus *bus) {
+    int r;
+    _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+    _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+    char** acquired = NULL;
+
+
+    r = sd_bus_call_method(
+            bus,
+            "org.freedesktop.DBus",
+            "/org/freedesktop/DBUS",
+            "org.freedesktop.DBus",
+            "ListNames",
+            &error,
+            &reply,
+            NULL);
+
+    if (r < 0) {
+        ZITI_LOG(ERROR, "Could not retreive DBus bus names: %s", error.message);
+	    return r;
+    }
+
+
+    r = sd_bus_message_read_strv(reply, &acquired);
+    if (r < 0) {
+        ZITI_LOG(ERROR, "Could not read dbus reply");
+        return r;
+    }
+
+    reply = sd_bus_message_unref(reply);
+
+
+    int i;
+    bool found = false;
+    for (i=0; acquired[i] != NULL; i++) {
+        if (strcmp(acquired[i], RESOLVED_BUS_NAME) == 0) {
+            ZITI_LOG(DEBUG, "systemd-resolved dbus name found: %s", acquired[i]);
+	    found = true;
+	    break;
+        }
+    }
+
+    if (acquired != NULL) {
+        for (i=0; acquired[i] != NULL; i++) {
+            free(acquired[i]);
+	}
+	free(acquired);
+    }
+
+    return found ? 0 : 1;
```

Comment:
why bother with boolean here - just make found an int and you can then just `return found`